CHANGELOG.md

```diff
@@ -1,5 +1,19 @@
 # Castle Core Changelog
 
+## Unreleased
+
+Deprecations:
+- The API surrounding `Lock` has been deprecated. This consists of the members listed below. Consider using the Base Class Library's `System.Threading.ReaderWriterLockSlim` instead. (@stakx, #391)
+   - `Castle.Core.Internal.Lock` (class)
+   - `Castle.Core.Internal.ILockHolder` (interface)
+   - `Castle.Core.Internal.IUpgradeableLockHolder` (interface)
+- The proxy type cache in `ModuleScope` should no longer be accessed directly. For this reason, the members listed below have been deprecated. (@stakx, #391)
+   - `Castle.DynamicProxy.ModuleScope.Lock` (property)
+   - `Castle.DynamicProxy.ModuleScope.GetFromCache` (method)
+   - `Castle.DynamicProxy.ModuleScope.RegisterInCache` (method)
+   - `Castle.DynamicProxy.Generators.BaseProxyGenerator.AddToCache` (method)
+   - `Castle.DynamicProxy.Generators.BaseProxyGenerator.GetFromCache` (method)
+
```

Comment:
Lets deprecate CacheKey too, since you're not supposed to use the cache directly!

Reply:
I'm already on it. :wink: